app/controllers/api/v4/runs_controller.rb

```diff
@@ -14,11 +14,30 @@ class Api::V4::RunsController < Api::V4::ApplicationController
   end
 
   def show
-    if params[:historic] == '1'
-      @run.parse(fast: false)
-      render json: @run, serializer: Api::V4::RunWithHistorySerializer
+    accept = request.headers.fetch('HTTP_ACCEPT', 'json').downcase
+    if accept == "original_timer"
+      accept = @run.program
+    end
+    timer = Run.program(accept)
+    if accept == 'json' || timer.nil? || !Run.exportable_programs.include?(timer)
+      if params[:historic] == '1'
+        @run.parse(fast: false)
+        render json: @run, serializer: Api::V4::RunWithHistorySerializer
+      else
+        render json: @run, serializer: Api::V4::RunSerializer
+      end
     else
-      render json: @run, serializer: Api::V4::RunSerializer
+      send_data(
+        if timer == Run.program(@run.timer)
+          @run.file
+        else
+          render_to_string(file: "runs/#{accept}.html.erb", layout: false)
+        end,
+        layout: false,
+        type: "application/#{accept}",
+        disposition: 'inline',
+        filename: "#{@run.filename(timer: timer)}"
+      )
     end
   end
```

Comment:
The extra conditionals and variables make this method a bit harder to follow now. Have you considered splitting it up into smaller functions?

Reply:
Is taking the if/else and filename call outside of the function call succinct enough?  Filename could probably be removed but I left it in case someone is testing something in browser and wanted to save the page, same with disposition.  As you said in another reply they most likely aren't completely needed, just kept for ease of use.

Reply:
Yeah filename should remain, I think the core readability issue with lots of functions in Splits I/O is having some inputs and then branching code below where multiple code paths use the inputs.

Inevitably the branches grow and when you're skimming logic that's 20+ lines away from the variable assignments that it's depending on, it's hard to understand at a glance what the values could be.

One possible solution is to condense the bigger of the branching code paths into methods, to end up with something like
```ruby
timer = format_from_headers(request.headers, run: @run)
if timer.present?
  render_as_format(timer) # a simple method that handles the send_data mess
  return
end

if params[:historic] == '1'
  @run.parse(fast: false)
  render json: @run, serializer: Api::V4::RunWithHistorySerializer
else
  render json: @run, serializer: Api::V4::RunSerializer
end
```